### PR TITLE
Add utils for saving relationships

### PR DESCRIPTION
**Pull request by bertdeblock**

### CODE REVIEW

File: addon/mixins/relationship-support-model.js [hunk 1]
```diff
@@ -2,10 +2,28 @@
...
 export default Mixin.create({
+  /**
+   * Hooks
+   */
+
+  init () {
+    this._super(...arguments)
+
+    deprecate(
+      'Use of the `relationship-support-model` mixin has been deprecated.',
```
Comment (Windvis) [inline addon/mixins/relationship-support-model.js]: Maybe mention the replacement util?

File: addon/mixins/relationship-support-model.js [hunk 2]
```diff
@@ -2,10 +2,28 @@
 
 import { RELATIONSHIP_ADAPTER_OPTION } from '@bagaaravel/ember-data-extensions/config'
 import getRelationshipDescriptor from '@bagaaravel/ember-data-extensions/utils/get-relationship-descriptor'
-import { assert } from '@ember/debug'
+import { assert, deprecate } from '@ember/debug'
 import Mixin from '@ember/object/mixin'
 
 export default Mixin.create({
```
Comment (Windvis) [inline addon/mixins/relationship-support-model.js]: Maybe you can refactor the mixin code to use the util internally? I know it's going to be deleted though.

File: addon/mixins/relationship-support-model.js (hunk 1 above)
Comment (bertdeblock) [reply addon/mixins/relationship-support-model.js]: I should mention them both then right?

File: addon/mixins/relationship-support-model.js (hunk 2 above)
Comment (bertdeblock) [reply addon/mixins/relationship-support-model.js]: Yeah, weird I forgot about this, f*ck was I thinking. It'll save some lines of code as well I think. Coming up!

File: addon/mixins/relationship-support-model.js (hunk 1 above)
Comment (Windvis) [reply addon/mixins/relationship-support-model.js]: Sure, or just refer to "utils" or something.

Comment (bertdeblock) [reply addon/mixins/relationship-support-model.js]: Done.

File: addon/mixins/relationship-support-model.js (hunk 2 above)
Comment (bertdeblock) [reply addon/mixins/relationship-support-model.js]: Done.

**Pull request merged**